### ISSUE TITLE
valid-jsdoc false positive for class documentation with class property arrow function

### ISSUE THREAD

**Issue by amannn** [eslint/eslint#9308]
**Tell us about your environment**

* **ESLint Version:** 4.6.1
* **Node Version:** 7.10.1
* **npm Version:** 4.2.0 (yarn 1.0.1)

**What parser (default, Babel-ESLint, etc.) are you using?**
babel-eslint

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "babel-eslint",
  "rules": {
    "valid-jsdoc": 2
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js
/**
 * Class description
 */
class Component {
  onClick = e => console.log(e);
}
```

**What did you expect to happen?**
No lint errors.

**What actually happened? Please include the actual, raw output from ESLint.**
At line number 1 the following errors are triggered:

 - `Missing JSDoc @returns for function. (valid-jsdoc)`
 - `Missing JSDoc for parameter 'e'. (valid-jsdoc)`

It looks like eslint thinks the comment above the class refers to the arrow function. If I replace the arrow function with a regular class method, the error vanishes. The error is also there when there's other methods above the class property function.

**Comment by not-an-aardvark** [comment]
Thanks for the report. However, class properties are experimental syntax, and we don't support experimental syntax in core rules. From our [readme](https://github.com/eslint/eslint/blob/12388d449f8d86358c22a2e39c9fe98bd3715823/README.md#what-about-experimental-features):

> ESLint doesn't natively support experimental ECMAScript language features. You can use babel-eslint to use any option available in Babel.

> Once a language feature has been adopted into the ECMAScript standard (stage 4 according to the TC39 process), we will accept issues and pull requests related to the new feature, subject to our contributing guidelines. Until then, please use the appropriate parser and plugin(s) for your experimental feature.

One option would be to create an issue on [babel-plugin-eslint](https://github.com/babel/eslint-plugin-babel) to see if they can create an alternative rule to resolve this.

**Comment by not-an-aardvark** [comment]
Closing because we don't support experimental syntax.

**Comment by amannn** [comment]
Oh, sorry I overlooked that. I'll open an issue over at babel-eslint then. Thanks!
